tool/templates/js2/runtime.mjs

```diff
@@ -0,0 +1,142 @@
+export function readString8(wasm, ptr, len) {
+	const buf = new Uint8Array(wasm.memory.buffer, ptr, len);
+	return (new TextDecoder("utf-8")).decode(buf)
+}
+
+export function readString16(wasm, ptr, len) {
+	const buf = new Uint16Array(wasm.memory.buffer, ptr, len);
+	return String.fromCharCode.apply(null, buf)
+}
+
+export function withWriteable(wasm, callback) {
+	const writeable = wasm.diplomat_buffer_writeable_create(0);
+	try {
+	callback(writeable);
+	const outStringPtr = wasm.diplomat_buffer_writeable_get_bytes(writeable);
+	const outStringLen = wasm.diplomat_buffer_writeable_len(writeable);
+	return readString8(wasm, outStringPtr, outStringLen);
+	} finally {
+	wasm.diplomat_buffer_writeable_destroy(writeable);
+	}
+}
+
+export class FFIError extends Error {
+	constructor(error_value) {
+	super("Error over FFI");
+	this.error_value = error_value; // (2)
+	}
+}
```

Comment:
note: I would like to get rid of this error wrapping in JS2. If a type is used in the error position of a `Result<T, E>`, it it should implement `Error` itself.

Reply:
That would be nice. I think it might require adding information to the HIR for Enums and Structs to see if they're ever used in that position

Reply:
Writing this since I just started encountering this with Result<> types, so I'm defaulting to the FFIError solution for now